junit-jupiter-api/src/main/java/org/junit/jupiter/api/Assertions.java

```diff
@@ -3448,4 +3448,48 @@ public static void assertTimeoutPreemptively(Duration timeout, Executable execut
 		return AssertTimeout.assertTimeoutPreemptively(timeout, supplier, messageSupplier);
 	}
 
+	/**
+	 * <em>Assert</em> that the supplied {@code actualValue} is an instance of the
+	 * {@code expectedType}.
+	 *
+	 * <p>Like the {@code instanceof} operator a {@code null} value is not
+	 * considered to be of the {@code expectedType} and does not pass the assertion.
+	 *
+	 * @since 5.8
+	 */
+	public static void assertInstanceOf(Class<?> expectedType, Object actualValue) {
```

Comment:
I wonder if the signature of this and the other methods shouldn't look like this to avoid obviously wrong parameters at compile time:

```suggestion
	public static <T> void assertInstanceOf(Class<? extends T> expectedType, T actualValue) {
```

Reply:
I don't think this is what you want. I found that in the examples I use it you don't know what actual type you really get so if you add the restriction you would need to add a cast the actual value which then fails instead of the assert getting us back to poor error messages.

Reply:
You're right, I had actually tried it and meant to remove this comment before posting the review. Sorry for the noise!